visualizer/src/Controls/ImageControls/RawControls/LayerOptions.js

```diff
@@ -0,0 +1,80 @@
+/** Modified from https://github.com/hms-dbmi/viv */
+import ClickAwayListener from '@material-ui/core/ClickAwayListener';
+import IconButton from '@material-ui/core/IconButton';
+import MenuItem from '@material-ui/core/MenuItem';
+import MenuList from '@material-ui/core/MenuList';
+import Paper from '@material-ui/core/Paper';
+import Popper from '@material-ui/core/Popper';
+import { makeStyles } from '@material-ui/core/styles';
+import MoreVertIcon from '@material-ui/icons/MoreVert';
+import { useSelector } from '@xstate/react';
+import React, { useReducer, useRef } from 'react';
+import { useRaw } from '../../../ServiceContext';
+import ColorPalette from './ColorPalette';
+
+const useStyles = makeStyles(() => ({
+  paper: {
+    backgroundColor: 'rgba(0, 0, 0, 0.75)',
+  },
+  span: {
+    width: '70px',
+    textAlign: 'center',
+    paddingLeft: '2px',
+    paddingRight: '2px',
+    color: 'white',
+  },
+  colors: {
+    '&:hover': {
+      backgroundColor: 'transparent',
+    },
+    paddingLeft: '2px',
+    paddingRight: '2px',
+  },
+}));
+
+function LayerOptions({ layer }) {
+  const [open, toggle] = useReducer(v => !v, false);
```

Comment:
Can you briefly comment on what this does? I'm not familiar with the reducer hook.

Reply:
The signature for `useReducer` is 
```const [state, dispatch] = useReducer(reducer, initialArg, init);```
so the `open` is the current state and `toggle` is a dispatch method with signature `(state, action) => newState` to update the state based on the current state and an action.

actions in reducers correspond to events for state machines, although there are no actions/events defined here, the reducer always does the same thing when it's dispatch method is called

You can think of a reducer as a state machine with just one state, i.e. the reducer always has the same behavior for an incoming events. State machines build on that model by allowing us to declare states that have different behaviors for each event and can transition between the states on the events. Compare `const [state, dispatch] = useReducer(reducer, initialState)` to `const [currentState, send] = useMachine(machine)`

When `toggle` is called, we take the current value of `open` and flip it. `open` starts as `false`, so the component is closed until it's clicked on